src/backend/cdb/cdbmutate.c

```diff
@@ -787,6 +797,136 @@ apply_motion_mutator(Node *node, ApplyMotionState *context)
 			return plan_tree_mutator(node, apply_motion_mutator, context);
 	}
 
+	/*
+	 * For UPDATE/DELETE, we check if there's any Redistribute or Broadcast
+	 * Motions before scan in the same subtree for the table we're going to
+	 * modify. If we encounter the scan before any motions, then we can elide
+	 * unneccessary Explicit Redistribute Motion.
+	 */
+	if (IsA(node, ModifyTable))
+	{
+		ModifyTable *mt = (ModifyTable *) node;
+
+		if (mt->operation == CMD_UPDATE || mt->operation == CMD_DELETE)
+		{
+			ListCell   *lcr;
+
+			context->mt.resultRelids = NIL;
+			context->mt.needExplicitMotion = false;
+			context->mt.nMotionsAbove = 0;
```

Comment:
Is this necessary? Since we initialize the structure in `apply_motion`. Can we add an Assert to the fact that the fields are in this state if we are not sure?

Reply:
Done. Added a comment that states the fact that we don't support multiple ModifyTable nodes in the same plan.